packages/typescript/index.ts

```diff
@@ -1099,18 +1101,57 @@ export class DagHandle {
             .then((response) => response.results);
     }
 
-    // public async dynamicList(
-    //     inputs: any[],
-    //     inputKey: string | undefined,
-    //     outputKey: string | undefined,
-    //     splitTh = 1000,
-    //     maxThreads = 50,
-    //     formatMethod: DynamicFormat.simple,
-    //     forceKeys = false,
-    //     noCache = false,
-    // ): Promise<any[]> {
-
-    // }
+    public async dynamicList(
+        inputs: any[],
+        inputKey: string = undefined,
+        outputKey: string = undefined,
+        splitTh: number | null = 1000,
+        formatMethod: DynamicFormat = 'simple',
+        forceKeys = false,
+        noCache = false
+    ): Promise<any[]> {
+        if (splitTh === null || inputs.length <= splitTh) {
+            return await this.client
+                .requestJSON<DynamicResults>({
+                    method: METHOD_POST,
+                    path: '/dynamic_list',
+                    args: {
+                        force_keys: forceKeys,
+                        format: formatMethod,
+                        input_key: inputKey,
+                        inputs: inputs,
+                        no_cache: noCache,
+                        output_key: outputKey,
+                        dag: this.getUri(),
+                    },
+                })
+                .then((res) => res.results);
+        }
+        const resArray: any[] = new Array(inputs.length).fill(null);
+        const splitNum = splitTh;
+        const computeHalf = async (cur: any[], offset: number) => {
+            if (cur.length <= splitNum) {
+                const curRes = await this.dynamicList(
+                    cur,
+                    isUndefined(inputKey) ? null : inputKey,
+                    isUndefined(outputKey) ? null : outputKey,
+                    null,
+                    formatMethod,
+                    forceKeys,
+                    noCache
+                );
+                Array.from(Array(curRes.length).keys()).forEach((ix) => {
+                    resArray[offset + ix] = curRes[ix];
+                });
+                return;
+            }
+            const halfIx = (cur.length / 2) >> 0;
+            await computeHalf(cur.slice(0, halfIx), offset);
+            await computeHalf(cur.slice(halfIx), offset + halfIx);
```

Comment:
dont' the awaits here force the computation? this looks like it would compute sequentially